feat: migrate datetime utility functions to v4 package (#1234)
contentful/forma-36: packages/components/datetime/DateTimeUtils.mdx

---
title: 'Utility functions'
type: 'component'
status: 'stable'
slug: /components/datetime-functions/
github: 'https://github.com/contentful/forma-36/tree/master/packages/components/datetime/src/utils'
typescript: ./src/utils/index.ts
---

> The `@contentful/f36-datetime` package provides a couple of funtions to format a date following [Forma 36 guidelines](https://f36.contentful.com/guidelines/date-and-time/).

Reviewer: ```suggestion
The `@contentful/f36-datetime` package provides a couple of functions to format a date following [Forma 36 guidelines](https://f36.contentful.com/guidelines/date-and-time/).
```

## formatDateAndTime

> This function will return a formatted string of any date that is passed as the first argument to the function.

Reviewer: ```suggestion
This function will return a formatted string of any date that is passed as the first argument of the function.
```

> It allows four different formats that can be chosen by passing one of the `format` options as a second argument to the function:

Reviewer: ```suggestion
It allows 4 different formats that can be chosen by passing one of the `format` options as a second argument of the function:
```

| `format` option | Result                   |
| --------------- | ------------------------ |
| `full`          | 2019-08-24T15:44:00.000Z |
| `day`           | 2019-08-24               |
| `time`          | 15:44:07.000             |
| `weekday`       | 08-24                    |

If no format is passed to the function, it will return a string with the `full` format

### Examples

```ts static=true
// passing a Timestamp as an ISOString
formatMachineReadableDateTime('2021-08-17T15:45');

// passing a JS Date
formatMachineReadableDateTime(new Date());

// passing a Unix Epoch in milliseconds
formatMachineReadableDateTime(1629240300000);
```

### Different formats

```ts static=true
// it will return "2021-08-17T15:44:00.000Z"
formatMachineReadableDateTime('2021-08-17T15:45', 'day');

// it will return "15:44:07.000 "
formatMachineReadableDateTime('2021-08-17T15:45', 'time');

// it will return "08-17"
formatMachineReadableDateTime('2021-08-17T15:45', 'weekday');
```

## formatRelativeDateTime

This function will return a string with how far a given date is in the past or future,

> using a baseDate as reference. If the baseDate is not passed, the function will use today as reference.

Reviewer: ```suggestion
using a `baseDate` as reference. If the `baseDate` is not passed, the function will use today as a reference.
```

### Examples

```ts static=true
// If today is 18.08.2021, it will return "a day ago"
formatRelativeDateTime('2021-08-17T15:45');

// If today is 18.08.2021, it will return "in a day"
formatRelativeDateTime('2021-08-19T15:45');

// it will return "in one week"
formatRelativeDateTime('2021-08-19T15:45', '2021-08-12T15:45');

// it will return "5 hours ago"
formatRelativeDateTime('2021-08-19T15:45', '2021-08-19T10:45');
```

## formatRelativeToCurrentWeekDateTime

> This function that formats a date relative to Today or to the `baseDate` if passed.

Reviewer: ```suggestion
This function formats a date relative to the current day or to the `baseDate` if passed.
```
